### ISSUE TITLE
Exempting certain classes/methods from various things

### ISSUE THREAD

**Issue by benjamin-kirkbride** [astral-sh/ruff#12076]
A lot of frameworks provide classes that have a set number of arguments. It is idiomatic to have them in place even when not used (ARG002), and they often have too many args (PLR0913).

An example of that from Pyglet:

```python
def on_mouse_drag(x, y, dx, dy, buttons, modifiers):
    pass
```

It is common to only need a couple of those arguments, but they all must be there for it to function properly. It is possible to `_` the args you don't need to get around ARG002, but that is not idiomatic and confuses thing.

Other examples exist that I don't have immediately on hand, things like Django class based views.

My proposal is a way to exempt classes (in the case of the init) and method names from these (and possibly other related?) checks.

Why not just disable those lints altogether? Because I want my own custom functions and methods to be held to a certain standard. Why not disable those checks on the functions where they come up? Because I will end up with dozens or hundreds of noqa comments throughout a medium sized program.

**Comment by fschulze** [comment]
Something along the lines of exceptions for mypy could also be helpful in many cases:

```python
def on_mouse_drag(x, y, dx, dy, buttons, modifiers):  # noqa: ARG002[x,y,buttons,modifiers]
    pass
```
The above should complain if ``dx`` or ``dy`` are unused, but not the others and it should fail if ``x``, ``y``, ``buttons`` or ``modifiers`` are used. That way one can be very specific and any changes will show up.
